### PR TITLE
Update the major version of Ktor 

### LINKED ISSUE #413: Update the major version of Ktor
### Issue description

Create next mayor branch for Micronaut Kotlin and update to next mayor version of Ktor

### PR DESCRIPTION
**Pull request by wetted**
closes #413 

This also fixes a prior [seemingly] bug where `MicronautKtorEnvironmentConfig.property(path)` erroneously prefixed `path` with just a dot character if `prefix` was empty (but not null), causing the call to `env.containsProperty` to fail. @graemerocher can you double check this for me please? I'm not 100% sure I understand how the new collection views `keys()` and `toMap()` should really look, with respect to `Environment`, not to mention my `Kotlin` kung-fu is a work in progress. ;)

### CODE REVIEW

**Comment by wetted** [comment]
Changes in core are also required:

https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/gradle/libs.versions.toml#L62

File: kotlin-runtime/src/main/kotlin/io/micronaut/kotlin/converters/FlowConverterRegistrar.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 @Requires(classes = {Flux.class, ReactiveFlowKt.class})
 public class FlowConverterRegistrar implements TypeConverterRegistrar {
     @Override
-    public void register(ConversionService<?> conversionService) {
+    public void register(MutableConversionService conversionService) {
```
Comment (timyates) [inline kotlin-runtime/src/main/kotlin/io/micronaut/kotlin/converters/FlowConverterRegistrar.java:37]:
```suggestion
    public void register(ConversionService conversionService) {
```

File: ktor/src/main/kotlin/io/micronaut/ktor/env/MicronautKtorEnvironmentConfig.kt [hunk 2]
```diff
@@ -43,23 +46,43 @@ class MicronautKtorEnvironmentConfig(val env : Environment, private val prefix :
...
     override fun property(path: String): ApplicationConfigValue {
-        val fullPath = if(prefix == null) path else "$prefix.$path"
-        if (env.containsProperty(fullPath)) {
-            return KtorApplicationConfigValue(fullPath, env)
+        val value = propertyOrNull(path)
+        if (value != null) {
+            return value
         } else {
             throw ApplicationConfigurationException("No configuration found for path: $path")
         }
```
Comment (timyates) [inline ktor/src/main/kotlin/io/micronaut/ktor/env/MicronautKtorEnvironmentConfig.kt:54]:
```suggestion
        return propertyOrNull(path) ?: throw ApplicationConfigurationException("No configuration found for path: $path")
```

**Review by timyates — COMMENTED** [review]
Not sure what the failure is...  It could be https://github.com/micronaut-projects/micronaut-core/issues/8419

File: kotlin-runtime/src/main/kotlin/io/micronaut/kotlin/converters/FlowConverterRegistrar.java (hunk 1 above)
Comment (wetted) [reply kotlin-runtime/src/main/kotlin/io/micronaut/kotlin/converters/FlowConverterRegistrar.java:37]: This won't compile. `addConverter()` is in the `MutableConversionService` interface

**Review by timyates — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
@wetted can you resolve the conflict?

**Comment by wetted** [comment]
The GraalVM job is failing with `API rate limit exceeded for 52.185.204.64`

**Comment by timyates** [comment]
@wetted Can you try applying https://github.com/micronaut-projects/micronaut-project-template/pull/309 here to the graal workflow?

**Pull request merged**